src/D2L.Bmx/AwsCredsCreator.cs

```diff
@@ -0,0 +1,65 @@
+using D2L.Bmx.Aws;
+using D2L.Bmx.Okta;
+
+namespace D2L.Bmx;
+
+internal class AwsCredsCreator {
+	private readonly IAwsClient _awsClient;
+	private readonly IConsolePrompter _consolePrompter;
+	private readonly BmxConfig _config;
+
+	public AwsCredsCreator(
+		IAwsClient awsClient,
+		IConsolePrompter consolePrompter,
+		BmxConfig config
+	) {
+		_awsClient = awsClient;
+		_consolePrompter = consolePrompter;
+		_config = config;
+	}
+
+	public async Task<AwsCredentials> CreateAwsCredsAsync(
+		IOktaApi oktaApi,
+		string? account,
+		string? role,
+		int? duration,
+		bool nonInteractive
+	) {
+		var accountState = await oktaApi.GetAccountsAsync( "amazon_aws" );
+		string[] accounts = accountState.Accounts;
+
+		if( string.IsNullOrEmpty( account ) ) {
+			if( !string.IsNullOrEmpty( _config.Account ) ) {
+				account = _config.Account;
+			} else if( !nonInteractive ) {
+				account = _consolePrompter.PromptAccount( accounts );
+			} else {
+				throw new BmxException( "Account value was not provided" );
+			}
+		}
+
+		string accountCredentials = await oktaApi.GetAccountAsync( accountState, account );
+		var roleState = _awsClient.GetRoles( accountCredentials );
+		string[] roles = roleState.Roles;
+
+		if( string.IsNullOrEmpty( role ) ) {
+			if( !string.IsNullOrEmpty( _config.Role ) ) {
+				role = _config.Role;
+			} else if( !nonInteractive ) {
+				role = _consolePrompter.PromptRole( roles );
+			} else {
+				throw new BmxException( "Role value was not provided" );
+			}
+		}
+
+		if( duration is null or 0 ) {
+			if( _config.DefaultDuration is not ( null or 0 ) ) {
```

Comment:
the lower bound is 15 minutes (https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use.html) we check against that as the lower bound instead of 0

Reply:
This is a check against some sort of "default" value that might exist when the user didn't explicitly specify anything, both null & 0 are plausible values for that, and we fall back to 60 in such cases.

If a user specifies 5 (or something crazier like -5), I'd think letting the process error out (and making the user aware) would be a more reasonable behaviour than implicitly increasing the value to 15.

We can opt to add explicit checks and error behaviour here, but out of my intended scope here.